recipes/fakeit/all/conanfile.py

```diff
@@ -0,0 +1,44 @@
+from conans import ConanFile, tools
+import os
+
+
+class FakeItConan(ConanFile):
+    name = "fakeit"
+    license = "MIT"
+    homepage = "https://github.com/eranpeer/FakeIt"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "C++ mocking made easy. A simple yet very expressive, headers only library for c++ mocking."
+    topics = ("mock", "fake", "spy")
+    settings = "compiler"
+    options = {
+        "integration": [None, "boost", "catch", "cute", "gtest", "mettle", "nunit", "mstest", "qtest", "standalone",
```

Comment:
Should we include those integration as requirements?

Reply:
Some of them aren't on conan center. And all integrations are included in the package, this option is only there to set the default integration used (the one that will be used by `<fakeit.hpp>` header, others have to be explicitly named with `<integration/fakeit.hpp>`.

So, we can do two things (i think): 
- we rename `integration` to `default_integration` to make it more clear and do nothing about the requirement, because requiring all 9 libraries all the time would be too heavy.
- we only include the integration named by the `integration` option into the package and add its library as a requirement.

The second option has the disadvantage of allowing only one integration at a time (and more work for creating the others packages), but I agree that it would be better.

Reply:
Or we can create two versions of each integrations, for example a `catch` that add a requirement to the catch package and a `noreq_catch` that doesn't. This way we would be able to add all integrations even the ones that don't have a package on conan center (they would be only available through the `noreq_` integration).

It has the disadvantage of making the package more complex (I don't really know what's the "standard" way of documenting options to the user when they're a bit complex, just a comment in the recipe is enough ?).

Reply:
We have in the past added requirements AFAIK

https://github.com/conan-io/conan-center-index/pull/3812/files#r537110951 <-- example we did

🤷 

IMO if the consumer needs to explicitly set an option they can easily set a requirement... but if this was compiled that would be different....

Will it be missing header includes? https://github.com/conan-io/conan-center-index/pull/4992/files#diff-dcefd0522786a3ad8878130e31083cfe9133f64cbdd09eafc309e5119a3f33d1R42-R44

Reply:
that depends. for instance, if you activate `-o fakeit:integration=boost` and it requires boost to compile/link/run, then boost clearly should be a requirement of fakeit.
but if  `-o fakeit:integration=boost` just makes it compatible with boost (e.g. it generates files containing boost includes), but doesn't require boost to compile/link/run, requirement could be omitted.

Reply:
Next PR for review preferences

Reply:
Uh, it was merged before I was able to resolve this conversation, so I'm not really sure what I should do now.

But the integrations headers requires the test framework dependency in order to compile. So if your only dependency is fakeit and you add `#include <catch/fakeit.hpp>` somewhere then your code won't compile anymore because `<catch/fakeit.hpp>` include the catch header that won't be present.

Reply:
Please open a second PR to add the build requirements

Reply:
I will but I can't do this change to the recipe without breaking compatibility, so it's a bit annoying it's already merged in master.

And what do I do about test frameworks that aren't in conan-center ? I don't provide the fakeit integration for them ?

Reply:
> without breaking compatibility

I believe it can be avoid, the options as they are are perfect fine

We have a common pattern

https://github.com/conan-io/conan-center-index/blob/4dc47f9ddbbe2c6f6d080fad5e9f3789a9c01ceb/recipes/getdns/all/conanfile.py#L83

And on rare occasions

https://github.com/conan-io/conan-center-index/blob/5ef66cb42668ff99e2d225636600b562c27ee6ed/recipes/zfp/all/conanfile.py#L58

I think that would be sufficient

Reply:
In the current version of the package all fakeit integration are included. It won't be the case once the requirements are added (because no one want to have 8 useless dependencies).

Ok for the warning message.